tests/workflow_contracts/mutation_testing_test.py

```
@@ -0,0 +1,144 @@
+
+def _triggers(workflow: dict) -> dict:
+    """Return the ``on:`` mapping (PyYAML parses the bare key as True)."""
+    triggers = workflow.get("on", workflow.get(True))
+    assert isinstance(triggers, dict), "the workflow must declare an on: mapping"
+    return triggers
+
+
+def _mutation_job(workflow: dict) -> dict:
+    """Return the single calling job."""
+    jobs = workflow.get("jobs")
+    assert isinstance(jobs, dict) and jobs, "the workflow must declare jobs"
```

Ruff (0.15.20)

[warning] Assertion should be broken down into multiple parts

(PT018)